AlicerceTests/ServiceLocatorTests.swift

```diff
@@ -0,0 +1,301 @@
+            
+        } catch ServiceLocatorError.inexistentService {
+            XCTAssert(true, "🎉 Service removed with success")
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testUnregisterUsingServiceName() {
+        
+        let testServiceString = "Test Register Service"
+        let serviceName = "ServiceName"
+        
+        do {
+            
+            // Register services
+            try serviceLocator.register(name: serviceName, service: testServiceString)
+            
+            // Get registered services
+            let rStringService: String = try serviceLocator.get(name: serviceName)
+            
+            // Check if it contains
+            XCTAssertEqual(rStringService, testServiceString)
+            
+            try serviceLocator.unregister(testServiceString, name: serviceName)
+            
+            let _: String = try serviceLocator.get(name: serviceName)
+            
+        } catch ServiceLocatorError.inexistentService {
+            XCTAssert(true, "🎉 Service removed with success")
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testUnregisterAll() {
+
+        let testServiceString = "Test Register Service"
+        let testServiceDouble = 1.0
+        let testLazyServiceInt: () -> (Int) = {
+            return 1
+        }
+
+        func assert(with error: Error) {
+            XCTAssert(false, "💥: UnregisterAll test failed with error: \(error)")
+        }
+
+        do {
+
+            // Register services
+            try serviceLocator.register(service: testServiceString)
+            try serviceLocator.register(service: testServiceDouble)
+            try serviceLocator.register(testLazyServiceInt)
+
+            // Check if all the services are loaded, even lazy ones
+            let rStringService: String = try serviceLocator.get()
+            let rDoubleService: Double = try serviceLocator.get()
+            let rIntService: Int = try serviceLocator.get()
+
+            // Validate that it contains registered services
+            XCTAssertEqual(rStringService, testServiceString)
+            XCTAssertEqual(rDoubleService, testServiceDouble)
+            XCTAssertEqual(rIntService, testLazyServiceInt())
+
+            serviceLocator.unregisterAll()
+
+        } catch let error {
+            assert(with: error)
+        }
+
+        func checkServiceRemoved<T>(_ type: T) {
+            do {
+                let _: T = try serviceLocator.get()
+
+            } catch ServiceLocatorError.inexistentService {
+                XCTAssert(true, "🎉 Service \(String(describing: type)) removed with success")
+            } catch let error {
+                assert(with: error)
+            }
+        }
+
+        // Test that services were all removed
+
+        checkServiceRemoved(Int.self)
```

Comment:
Already discussed this in person. 💪 

OK, i agree with one dict for  the services but not the api for `list()` i think we don't need it.
Regarding the `enum`, we discussed we don't need the service `Type` for validation.